pkg/devfile/image/image.go

```diff
@@ -89,7 +90,7 @@ func buildPushImage(backend Backend, image *devfile.ImageComponent, devfilePath
 // selectBackend selects the container backend to use for building and pushing images
 // It will detect podman and docker CLIs (in this order),
 // or return an error if none are present locally
-func selectBackend() (Backend, error) {
+func selectBackend(fs filesystem.Filesystem) (Backend, error) {
```

Comment:
Don't find this change comfortable. Actually `selectBackend` has no need for the `fs`. It's doing something else along with trying to figure out the backend to use, and hence it now needs this extra param.

Reply:
Makes sense. Instead, I moved this parameter to the `Build` method in the `image.Backend` interface. This is where it is actually needed, when trying to build the image.
See [`878285b` (#5976)](https://github.com/redhat-developer/odo/pull/5976/commits/878285b711c77ade4cf2e7ac7f12d269d364046e)